girder_worker_utils/transforms/girder_io.py

```diff
@@ -28,9 +28,21 @@ def __init__(self, *args, **kwargs):
                     # Fall back if the worker plugin is unavailble
                     except ImportError:
                         from girder.api.rest import getApiUrl as getWorkerApiUrl
-                from girder.api.rest import getCurrentToken
+
                 self.gc = GirderClient(apiUrl=getWorkerApiUrl())
-                self.gc.token = getCurrentToken()['_id']
+                from girder.api.rest import getCurrentUser
+                if getCurrentUser():
+                    from girder.constants import TokenScope
+                    from girder.models.token import Token
+                    token = Token().createToken(
+                        days=1,
+                        scope=[TokenScope.DATA_READ, TokenScope.DATA_WRITE],
```

Comment:
My concern with using this scope is that it might not allow writes back to the job itself. I could be wrong about that though... Have you tried it?

Reply:
Yes, I've tested this and it writes the status back to the job just fine.

Reply:
Great. I sanity checked the relevant code and that makes sense since this is a user-associated token:

```python
    def updateJob(self, job, log, overwrite, notify, status, progressTotal, progressCurrent,
                  progressMessage):
        user = self.getCurrentUser()
        if user:
            self._model.requireAccess(job, user, level=AccessType.WRITE)
        else:
            self.ensureTokenScopes('jobs.job_' + str(job['_id']))
```